reference/3.0/Microsoft.PowerShell.Core/About/about_Ref.md

```diff
@@ -5,72 +5,135 @@ locale:  en-us
 keywords:  powershell,cmdlet
 title:  about_Ref
 ---
+
 # About Ref
 
-## SHORT DESCRIPTION
+## Short description
 
-Describes how to create and use a reference variable type.
+Describes how to create and use a reference variable type. You can use
+reference variables type to permit a method to change the value
+of a variable that is passed to it.
 
-## LONG DESCRIPTION
+## Long description
 
-You can use the reference variable type to permit a method to change the value
-of a variable that is passed to it.
+You can pass variables to methods *by reference* or *by value*.
 
-When the [ref] type is associated with an object, it returns a reference to
-that object. If the reference is used with a method, the method can refer to
-the object that was passed to it. If the object is changed within the method,
-the change appears as a change in the value of the variable when control
-returns to the calling method.
+In the following example, a method attempts to change the value of a
+variable that is passed into it, and fails.
+
+```powershell
+Function Test($data)
+{
+    $data = 3
+}
+
+$var = 10
+Test -data $var
+$var
+```
 
-To use referencing, the parameter must be a reference variable. If it is not,
-an InvalidArgument exception is thrown.
+```output
+10
+```
 
-The parameters used in method invocations must match the type required by the
-methods.
+When you pass a variable *by value*, you are passing a copy of the data.
 
-Examples:
+In the following example, a variable containing a `Hashtable` is passed to a
+method. The method adds its own key, which you can still view after the
+method executes.
 
 ```powershell
-PS> function swap([ref]$a,[ref]$b)
->> {
->>     $a.value,$b.value = $b.value,$a.value
->> }
-
-PS> $a = 1
-PS> $b = 10
-PS> $a,$b
-1
-10
-PS> swap ([ref]$a) ([ref]$b)
-PS> $a,$b
-10
-1
-
-PS C:\ps-test> function double
->> {
->>     param ([ref]$x) $x.value = $x.value * 2
->> }
-
-PS> $number = 8
-PS> $number
-8
-PS C> double ([ref]$number)
-PS C> $number
-16
+Function Test($data)
+{
+    $data.Value = "New Text"
+}
+
+$var = @{}
+Test -data $var
+$var.Value
```

Comment:
Using .Value here is confusing because you are creating a Key named 'Value' that has a Value property associated with it. Let's choose a different name.

Reply:
fixed